.sandbox/provision/lifecycle-asg/on-delete.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+source ./common.sh
+
+redirect_output on-delete.log
+
+INSTANCE_ID=$(stored_instance_id)
+VOLUME_ID=$(stored_volume_id)
+
+if [ -n "$INSTANCE_ID" ] ; then
+    aws ec2 terminate-instances --instance-ids $INSTANCE_ID
+    aws ec2 wait instance-terminated --instance-ids $INSTANCE_ID
+fi
+
+if [ -n "$VOLUME_ID" ]; then
+    aws ec2 delete-volume --volume-id $VOLUME_ID
+fi
+
+restore_output
```

Comment:
I'd recommend using tags all the time and not saving the IDs. So we can reduce the state and even eliminate that. The tags can be derived from unique information in the environment, like sandbox ID.

Reply:
Given the 0 maintenance cost of the state file, I could not see much benefit to do extra complex queries to the AWS API in order to retrieve something that we already know. Would you mind sheding more lights on this?

Reply:
Adding a state file adds extra complexity. You don't know what will happen to the file. The complexity is not about the code, but the runtime situation.